File: springwebfluxtutorial/src/main/java/com/qiusuo/webfluxtutorial/service/BookService.java
package com.qiusuo.webfluxtutorial.service;

import com.qiusuo.webfluxtutorial.domain.Book;
import com.qiusuo.webfluxtutorial.repository.BookRepository;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;
import java.util.concurrent.CompletableFuture;

@Service
public class BookService {
    private BookRepository bookRepository;

    public BookService(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    public Mono<Book> createBook(String isoNumber, String title) {
        Book book = new Book(isoNumber, title);
        return this.bookRepository.save(book);
    }
    
    public Flux<Book> findAll() {
        return bookRepository.findAll();
    }
}

File: springwebfluxtutorial/src/main/kotlin/com/qiusuo/webfluxtutorial/service/BookService.kt
package com.qiusuo.webfluxtutorial.service

import com.qiusuo.webfluxtutorial.domain.Book
import com.qiusuo.webfluxtutorial.repository.BookRepository
import org.springframework.stereotype.Service
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Service
class BookService(private val bookRepository: BookRepository) {
    fun createBook(isoNumber: String?, title: String?): Mono<Book> {
        val book = Book(isoNumber,
            title!!, true
        )
        return bookRepository.save(book)
    }

    fun findAll(): Flux<Book> {
        return bookRepository.findAll()
    }
}